### PR TITLE
feat(new): Unreal Cloud DDC

### LINKED ISSUE #277: [Module Proposal]: Unreal Cloud DDC
### Have you checked this module does not already exist in the registry?

Yes

### Why is the module needed?

Make it easier for Game Developers to deploy the set of resources required to run Unreal Cloud DDC on Azure.

### Module path

azure-gaming/unreal-cloud-ddc

### Describe the module

This uses AKS, CosmosDB, and Azure Storage to host the DDC assets of a developer in the cloud which can be used by other members of the game studio.

### PR DESCRIPTION
**Pull request by dciborow**
## Description
Closes #277 

This PR features nested templates in the `modules` directory which we are working to replace but are still pending the following PRs. The versions of those templates in this PR made be out of date and not reflect the latest feedback. As the following PRs are approved, we will update this template with the latest changes.

https://github.com/Azure/bicep-registry-modules/pull/254
https://github.com/Azure/bicep-registry-modules/pull/251
https://github.com/Azure/bicep-registry-modules/pull/252
https://github.com/Azure/bicep-registry-modules/pull/253

- [ ] #251 
- [ ] #252
- [ ] #253 
- [ ] #254 

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

### CODE REVIEW

**Comment by dciborow** [comment]
@shenglol , there is a parameter that I need to provide for the test case which I would not like to be public (URL to our private container registry). 
Is there a way that we can wire this up using a repo secret?

**Review by manuvar-ms — APPROVED** [review]

File: modules/azure-gaming/unreal-cloud-ddc/modules/ContainerService/aks-run-command.bicep [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+param cleanupPreference string = 'OnSuccess'
+
+@description('Set to false to deploy from as an ARM template for debugging') 
+param isApp bool = true
+
+resource aks 'Microsoft.ContainerService/managedClusters@2022-01-02-preview' existing = {
+  name: aksName
+}
+
+resource newDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = if (!useExistingManagedIdentity) {
```
Comment (Gordonby) [inline modules/azure-gaming/unreal-cloud-ddc/modules/ContainerService/aks-run-command.bicep:49]: you could bump this API version to be much newer (same with some other lines) :)

File: .github/workflows/gpt-pr-review.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    permissions: write-all
+    runs-on: ubuntu-latest
+    name: OpenAI PR Comment
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 2
+      - id: review
+        uses: dciborow/action-gpt@dciborow/static-analysis
```
Comment (shenglol) [inline .github/workflows/gpt-pr-review.yml:14]: @dciborow What does the action do? I don't feel like this is something we should check in this now without having a discussion with the team.

Comment (dciborow) [reply .github/workflows/gpt-pr-review.yml:14]: this snuck in by mistake from my fork.

File: modules/azure-gaming/unreal-cloud-ddc/test/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/*
+Write deployment tests in this file. Any module that references the main
+module file is a deployment test. Make sure at least one test is added.
+*/
+param location string = resourceGroup().location
+param helmChart string
```
Comment (dciborow) [inline modules/azure-gaming/unreal-cloud-ddc/test/main.test.bicep:6]: @shenglol , i need to set this value to a secret, any suggestions?

**Pull request closed without merging**